pkg/controller/utils/kubeproxy.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	KubeProxyInstanceKey = client.ObjectKey{Name: "kube-proxy", Namespace: "kube-system"}
+)
+
+// ValidateDaemonSetManaged checks whether the DaemonSet is managed by an external tool,
+// based on common labels or annotations typically added by automation.
+func ValidateDaemonSetManaged(ds *appsv1.DaemonSet) error {
```

Comment:
I think this name should be changed to `ValidateDaemonSetUnmanaged` because no error is the positive result for the function.